### DISCUSSION TITLE
CBCT using DICOM

### DISCUSSION THREAD

**Discussion by arvindmvepa** [astra-toolbox/astra-toolbox#340] [General]
I was having some issues using the Python Astral toolbox for 3D Reconstruction of rotational angiography . The rotational angiography is stored in a DICOM format. I used `astra.create_proj_geom('cone',detector_spacing_x, detector_spacing_y, det_row_count, det_col_count, angles, source_origin, origin_det)` to create the projection geometry and associated the method parameters with the DICOM attributes as follows:

detector_spacing_* (float) – Imager Pixel Spacing [x,y] from DICOM
det_row_count (int) – # of rows in projection slice
det_col_count (int) – # of cols in projection slice
angles (numpy.ndarray) – Using the "Positioner Primary Angle" in DICOM as a base angle and then incrementally adding the Positioner Primary Angle Increment in DICOM to generate an array of angles. Then convert from degrees to radians.
source_origin (float) –  Distance Source to Patient in DICOM
origin_det (float) – Distance Source to Detector in DICOM - Distance Source to Patient in DICOM

Then, I use  `astra.data3d.link("-sino", proj_geom, arr)` to link the DICOM pixel array to the projection geometry. After doing this, I essentially follow https://github.com/astra-toolbox/astra-toolbox/blob/fa2ec619edb2994e828897e80c06e7fb35c55c44/samples/python/s007_3d_reconstruction.py to apply a 3D Reconstruction algorithm based on the stored projection data and generate the 3D reconstruction. I tried BP3D_CUDA, FDK_CUDA, CGLS3D_CDA, and SIRT3D_CUDA to generate the 3D reconstruction. However, I was unable to achieve good results. Do you have any thoughts on what my issues are?

**Comment by askorikov** [comment]
Hi! All steps you described seem reasonable. In what sense is the reconstruction not good?
